.cirrus.yml

```diff
@@ -12,19 +12,25 @@ prep_stuff_template: &PREP_STUFF_TEMPLATE
       - cat pytest.ini | grep "addopts = " | cut -d'=' -f2 |  sed 's/--/+/g' | tr '+' '\n' | grep bitcoin |  cut -d' ' -f2 
       - cat tests/bitcoin_gitrev_pinned 2> /dev/null || true
       - cat /etc/os-release | grep VERSION
+      - cat ./tests/install_noded.sh
+      - echo "binary" # if the next line is --binary, otherwise compile. Not doing that correctly will break a proper caching switching between both
     populate_script: ./tests/install_noded.sh --debug --bitcoin binary
   elementsd_installation_cache:
     folder: ./tests/elements
     fingerprint_script: 
       - cat pytest.ini | grep "addopts = " | cut -d'=' -f2 |  sed 's/--/+/g' | tr '+' '\n' | grep elements |  cut -d' ' -f2 
       - cat tests/elements_gitrev_pinned 2> /dev/null || true
       - cat /etc/os-release | grep VERSION
+      - cat ./tests/install_noded.sh
+      - echo "binary" # if the next line is --binary, otherwise compile. Not doing that correctly will break a proper caching switching between both
```

Comment:
```suggestion
      - echo "binary" # if the next line is --elements binary, otherwise use echo "compile" - this ensures different caching keys.
```